tests/integration/build_test.sh

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env bash
+
+include './src/lib/kwio.sh'
+include './src/lib/kwlib.sh'
+include './tests/unit/utils.sh'
+include './tests/integration/utils.sh'
+
+declare -g CLONED_KERNEL_TREE_PATH_HOST
+declare -g TARGET_RANDOM_DISTRO
+declare -g KERNEL_TREE_PATH_CONTAINER
+declare -g CONTAINER
+
+function oneTimeSetUp()
+{
+  local url_kernel_repo_tree='https://github.com/torvalds/linux'
+
+  # Select a random distro for the tests
+  TARGET_RANDOM_DISTRO=$(select_random_distro)
+  CLONED_KERNEL_TREE_PATH_HOST="$(mktemp --directory)/linux"
+  CONTAINER="kw-${TARGET_RANDOM_DISTRO}"
+
+  # The VERBOSE variable is set and exported in the run_tests.sh script based
+  # on the command-line options provided by the user. It controls the verbosity
+  # of the output during the test runs.
+  setup_container_environment "$VERBOSE" 'build' "$TARGET_RANDOM_DISTRO"
+
+  # Install kernel build dependencies
+  container_exec "$CONTAINER" 'yes | ./setup.sh --install-kernel-dev-deps'
+  if [[ "$?" -ne 0 ]]; then
+    complain "Failed to install kernel build dependencies for ${TARGET_RANDOM_DISTRO}"
+    return 22 # EINVAL
+  fi
+
+  git clone --depth 5 "$url_kernel_repo_tree" "$CLONED_KERNEL_TREE_PATH_HOST"
+  if [[ "$?" -ne 0 ]]; then
+    complain "Failed to clone ${url_kernel_repo_tree}"
+    if [[ -n "$CLONED_KERNEL_TREE_PATH_HOST" ]]; then
+      if is_safe_path_to_remove "$CLONED_KERNEL_TREE_PATH_HOST"; then
+        rm -rf "$CLONED_KERNEL_TREE_PATH_HOST"
+      else
+        complain "Unsafe path: ${CLONED_KERNEL_TREE_PATH_HOST} - Not removing"
+      fi
+    else
+      complain 'Variable CLONED_KERNEL_TREE_PATH_HOST is empty or not set'
+    fi
+  fi
+}
+
+function setUp()
+{
+  KERNEL_TREE_PATH_CONTAINER="$(container_exec "$CONTAINER" 'mktemp --directory')/linux"
+  if [[ "$?" -ne 0 ]]; then
+    fail "(${LINENO}): Failed to create temporary directory in container."
+  fi
+}
+
+function setup_kernel_tree_with_config_file()
```

Comment:
I think you should call this function inside the `setUp` and remove it from the specific tests.